packages/ember-auto-import/ts/auto-import.ts

```diff
@@ -16,7 +17,7 @@ const debugTree = buildDebugCallback('ember-auto-import');
 // what you're doing.
 export interface AutoImportSharedAPI {
   isPrimary(addonInstance: AddonInstance): boolean;
-  analyze(tree: Node, addon: AddonInstance): Node;
+  analyze(tree: Node, addon: AddonInstance, treeType?: TreeType): Node;
```

Comment:
This area deserves extra scrutiny because different versions of ember-auto-import will call each other's copy of `analyze`. But this looks safe to me -- older copies will not pass the optional argument, newer copies will pass the argument and it will be ignored.